File: 1- Two Sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dict = {}
        for i, num in enumerate(nums):
            if target - num in dict:
                return [dict[target - num], i]
            dict[num] = i


'''
设置一个map容器用来记录元素的key:值与velue:索引
'''

File: C++/1-Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> backup(nums);
        sort(nums.begin(), nums.end());
        vector<int>::iterator left=nums.begin(), right=nums.end() - 1;
        while(*left + *right != target){
            while(*left + *right < target) left++;
            while(*left + *right > target) right--;
        }
        vector<int> result;
        for (int i = 0; i < nums.size(); i++) {
            if (backup[i] == *left || backup[i] == *right) result.push_back(i);
        }
        return result;
    }
};
